akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/EventSourcedBehaviorTestKitImpl.scala

```diff
@@ -150,10 +132,74 @@ import akka.persistence.typed.internal.EventSourcedBehaviorImpl
       replyProbe.stop()
     }
 
+    val newState = getState()
```

Comment:
Most of the changes are refactoring to extract the common code to private methods. Here is what makes the async replies work. It will first wait for the reply and then retrieve state and events.